### PR TITLE
Api japan improvements

### PR DESCRIPTION
**Pull request by dknochen**

### CODE REVIEW

File: queries/public/api/facts.jq [hunk 1]
```diff
@@ -53,17 +55,29 @@ declare function local:param-values($name as string) as string*
                then $companies
                else "dummy",
         request:param-values("xbrl:Entity"))
-     case $name eq "xbrl28:Archive" and $profile-name eq "sec" return (
-            let $fiscalYears := ($fiscalYear, request:param-values("sec:FiscalYear"))
-            let $fiscalPeriods := local:param-values("sec:FiscalPeriod")
+     case $name eq "xbrl28:Archive" and $profile-name = ("sec", "japan") return (
+            let $prefix as string := ("sec"[$profile-name eq "sec"], "fsa"[$profile-name eq "japan"])
```
Comment (gislenius) [inline queries/public/api/facts.jq]: I think a switch would be more appropriate here.

**Comment by gislenius** [comment]
Hi Dennis,

It looks very good! Just a minor comment and then I'll approve.

File: queries/public/api/facts.jq (hunk 1 above)
Comment (dknochen) [inline queries/public/api/facts.jq]: done

**Comment by dknochen** [comment]
Hi Ghislain,
Addressed your comments.
thank you.

**Comment by gislenius** [comment]
Thanks!

**Pull request merged**